packages/application/src/shell.ts

```diff
@@ -1993,11 +2021,35 @@ namespace Private {
       const collapsed = this._sideBar.currentTitle === null;
       const widgets = Array.from(this._stackedPanel.widgets);
       const currentWidget = widgets[this._sideBar.currentIndex];
+      let sizes = new Array<number>();
+      let expansionStates = new Array<boolean>();
+      Object.entries(this).forEach(([key, value]) => {
```

Comment:
I don't think this is doing what you want. You should loop on `this._stackedPanel.widgets` not `this` I guess. And then test if `panel.content` is an instance of `SplitPanel` (you don't need to test for `AccordionPanel` as it inherits from `SplitPanel`).

> Tip: you can directly use the local `widgets` variables.